### PR TITLE
feat: Example of AWS CodeCommit as a repository for ArgoCD workloads

### PR DESCRIPTION
**Pull request by ybezsonov**
### What does this PR do?

This example shows how to provision an EKS cluster with:
- ArgoCD
  - Workloads and addons deployed by ArgoCD (main.ft, locals.tf, providers.tf, outputs.tf, data.tf)
  - Private AWS CodeCommit repository and AWS Lambda stack provisioned by included Terraform script (repo-cc.tf) and used for workloads
  - AWS Lambda triggers ArgoCD projects sync using ArgoCD webhook after each push to the AWS CodeCommit repository

- [ Yes] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [Yes ] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [ Yes] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [ ] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs require a new example and/or doc page. In general:
- Use an existing example when possible to demonstrate a new addons usage
- A new docs page under `docs/add-ons/*` is required for new a new addon

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: examples/gitops/argocd-codecommit/README.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+- [Documentation](https://aws-ia.github.io/terraform-aws-eks-blueprints/latest/add-ons/argocd/)
+- [EKS Blueprints Add-ons Repo](https://github.com/aws-samples/eks-blueprints-add-ons)
+- [EKS Blueprints Workloads Repo](https://github.com/aws-samples/eks-blueprints-workloads)
+- [EKS Blueprints for Terraform Workshop](https://catalog.workshops.aws/eks-blueprints-terraform/en-US)
+
+## Prerequisites
+
+Main Terraform files (main.ft, locals.tf, providers.tf, outputs.tf, data.tf) were created using instructions from [EKS Blueprints for Terraform Workshop](https://catalog.workshops.aws/eks-blueprints-terraform/en-US)
+Solution specifc sript is repo-cc.tf
```
Comment (allamand) [inline examples/gitops/argocd-codecommit/README.md:24]: Specific

File: examples/gitops/argocd-codecommit/README.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+Ensure that you have the following tools installed locally:
+
+1. [aws cli](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html)
+2. [kubectl](https://Kubernetes.io/docs/tasks/tools/)
+3. [terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli)
+
+Change the link of GitHub repository for workloads from [EKS Blueprints for Terraform Workshop](https://catalog.workshops.aws/eks-blueprints-terraform/en-US) to AWS CodeCommit repository provisioned by the stack in repo-cc.tf
```
Comment (allamand) [inline examples/gitops/argocd-codecommit/README.md:32]: Should be the link for the GitHub workload repo ?

**Review by allamand — COMMENTED** [review]
Argo can’t be triggered directly on push in code commit without a lambda ?

File: examples/gitops/argocd-codecommit/README.md (hunk 1 above)
Comment (ybezsonov) [reply examples/gitops/argocd-codecommit/README.md:24]: Argo is triggered by webhook. Github does it out of the box and for Codecommit additional setup is required: trigger and lambda to call ArgoCD webhook url.
The solution will also work without trigger and webhook, but it this case Argo will use default periodic (every X minutes) calls to CodeCommit repo to check if something changed.
Trigger and Lambda is an example of alternative approach which can speed up feedback for developer and remove unnecessary sync calls.

File: examples/gitops/argocd-codecommit/README.md (hunk 2 above)
Comment (ybezsonov) [reply examples/gitops/argocd-codecommit/README.md]: fixed

File: examples/gitops/argocd-codecommit/README.md (hunk 1 above)
Comment (ybezsonov) [reply examples/gitops/argocd-codecommit/README.md]: > Specific

fixed

**Comment by bryantbiggs** [comment]
thank you for the PR @ybezsonov - lets start with an issue first and see if there is interest for this type of blueprint first. We can always re-open this PR in the future if there is

**Pull request closed without merging**